server/src/main/java/com/thoughtworks/go/config/ConfigConverter.java

```diff
@@ -639,4 +641,395 @@ private TrackingTool toTrackingTool(CRTrackingTool crTrackingTool) {
+
+        if (task instanceof PluggableTask)
+            return pluggableTaskToCRPluggableTask((PluggableTask) task);
+        else if (task instanceof BuildTask) {
+            return buildTaskToCRBuildTask((BuildTask) task);
+        } else if (task instanceof ExecTask) {
+            return execTasktoCRExecTask((ExecTask) task);
+        } else if (task instanceof FetchTask) {
+            return fetchTaskToCRFetchTask((FetchTask) task);
+        } else if (task instanceof FetchPluggableArtifactTask) {
+            return fetchPluggableArtifactTaskToCRFetchPluggableTask((FetchPluggableArtifactTask) task);
+        } else
+            throw new RuntimeException(
+                    String.format("unknown type of task '%s'", task));
+    }
+
+    public CRFetchPluggableArtifactTask fetchPluggableArtifactTaskToCRFetchPluggableTask(FetchPluggableArtifactTask task) {
+        List<CRConfigurationProperty> configuration = configurationToCRConfiguration(task.getConfiguration());
+        CRFetchPluggableArtifactTask crTask = new CRFetchPluggableArtifactTask(
+                task.getStage().toString(),
+                task.getJob().toString(), task.getArtifactId(), configuration);
+        crTask.setPipelineName(task.getPipelineName().toString());
+        commonCRTaskMembers(crTask, task);
+        return crTask;
+    }
+
+    public CRFetchArtifactTask fetchTaskToCRFetchTask(FetchTask task) {
+        CRFetchArtifactTask fetchTask = new CRFetchArtifactTask(
+                task.getStage().toString(),
+                task.getJob().toString(),
+                task.getSrc());
+
+        fetchTask.setDestination(task.getDest());
+        fetchTask.setPipelineName(task.getPipelineName().toString());
+        fetchTask.setSourceIsDirectory(task.isSourceAFile());
+
+        commonCRTaskMembers(fetchTask, task);
+        return fetchTask;
+    }
+
+    public CRPluggableTask pluggableTaskToCRPluggableTask(PluggableTask pluggableTask) {
+        CRPluginConfiguration pluginConfiguration = new CRPluginConfiguration(pluggableTask.getPluginConfiguration().getId(), pluggableTask.getPluginConfiguration().getVersion());
+        List<CRConfigurationProperty> configuration = configurationToCRConfiguration(pluggableTask.getConfiguration());
+        CRPluggableTask task = new CRPluggableTask(pluginConfiguration, configuration);
+        commonCRTaskMembers(task, pluggableTask);
+        return task;
+    }
+
+    public CRExecTask execTasktoCRExecTask(ExecTask task) {
+        CRExecTask crExecTask = new CRExecTask(task.getCommand());
+        crExecTask.setTimeout(task.getTimeout());
+        crExecTask.setWorkingDirectory(task.workingDirectory());
+        crExecTask.setArgs(task.getArgList().stream()
+                .map(Argument::getValue)
+                .collect(Collectors.toList())
+        );
+
+        commonCRTaskMembers(crExecTask, task);
+        return crExecTask;
+    }
+
+    public CRBuildTask buildTaskToCRBuildTask(BuildTask buildTask) {
```

Comment:
Can you make all of these methods private/package-private if they don't need to be public? IntelliJ shows that as a warning.